modules/setting/setting.go

```diff
@@ -643,6 +631,21 @@ please consider changing to GITEA_CUSTOM`)
 	Domain = sec.Key("DOMAIN").MustString("localhost")
 	HTTPAddr = sec.Key("HTTP_ADDR").MustString("0.0.0.0")
 	HTTPPort = sec.Key("HTTP_PORT").MustString("3000")
+
+	defaultAppURL := string(Protocol) + "://" + Domain + ":" + HTTPPort + "/"
```

Comment:
That's a good suggestion.  I had just recreated what was in conf/app.ini, but your refinement makes more sense and is probably what everyone would expect to happen.

I'll make that change.